internal/monitors/logstash/logstash/monitor.go

```diff
@@ -0,0 +1,232 @@
+		dps = append(dps, m.fetchPluginMetrics(client, fmt.Sprintf("%s://%s:%d%s", scheme, m.conf.Host, m.conf.Port, pluginPath), "node.plugins", dims)...)
+
+		now := time.Now()
+		for _, dp := range dps {
+			dp.Timestamp = now
+			m.Output.SendDatapoint(dp)
+		}
+
+	}, time.Duration(conf.IntervalSeconds)*time.Second)
+
+	return nil
+}
+
+func (m *Monitor) fetchNodeInfo(client *http.Client, endpoint string) map[string]string {
+	dims := make(map[string]string)
+
+	if nodeJSON, err := getJSON(client, endpoint); err == nil {
+		if nodeID, exists := nodeJSON["id"]; exists {
+			dims["node_id"], _ = nodeID.(string)
+		}
+		if nodeName, exists := nodeJSON["name"]; exists {
+			dims["node_name"], _ = nodeName.(string)
+		}
+	}
+
+	return dims
+}
+
+func (m *Monitor) fetchMetrics(client *http.Client, endpoint string, prefix string, dimensions map[string]string) []*datapoint.Datapoint {
+	var dps []*datapoint.Datapoint
+
+	if metricsJSON, err := getJSON(client, endpoint); err == nil {
+		dps = m.extractDatapoints(prefix, metricsJSON, dimensions)
+	}
+
+	return dps
+}
+
+func (m *Monitor) fetchPipelineMetrics(client *http.Client, endpoint string, prefix string, dimensions map[string]string) []*datapoint.Datapoint {
+	var dps []*datapoint.Datapoint
+
+	if metricsJSON, err := getJSON(client, endpoint); err == nil {
+		if pipelines, exists := metricsJSON["pipelines"]; exists {
+			if pipelinesObj, isObject := pipelines.(map[string]interface{}); isObject {
+				for pipelineName, pipelineObj := range pipelinesObj {
+					if pipeline, converted := pipelineObj.(map[string]interface{}); converted {
+						dimsClone := make(map[string]string)
+						for dimk, dimv := range dimensions {
+							dimsClone[dimk] = dimv
+						}
+						dimsClone["pipeline"] = pipelineName
+						dps = append(dps, m.extractDatapoints(prefix, pipeline, dimsClone)...)
+					}
+				}
+			}
+		}
+	}
+
+	return dps
+}
+
+func (m *Monitor) fetchPluginMetrics(client *http.Client, endpoint string, prefix string, dimensions map[string]string) []*datapoint.Datapoint {
+	if metricsJSON, err := getJSON(client, endpoint); err == nil {
+		if total, exists := metricsJSON["total"]; exists {
+			metricName := fmt.Sprintf("%s.%s", prefix, "total")
+			metricType, isEnabled := m.metricTypeMap[metricName]
+			metricValue, castErr := datapoint.CastMetricValueWithBool(total)
+
+			if isEnabled && castErr == nil {
+				return []*datapoint.Datapoint{
+					&datapoint.Datapoint{
+						Metric:     metricName,
+						MetricType: metricType,
+						Value:      metricValue,
+						Dimensions: dimensions,
+					},
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (m *Monitor) extractDatapoints(metricPath string, metricsJSON map[string]interface{}, dims map[string]string) []*datapoint.Datapoint {
+	var dps []*datapoint.Datapoint
+
+	for k, v := range metricsJSON {
+		childPath := fmt.Sprintf("%s.%s", metricPath, k)
```

Comment:
Probably going to be faster to do `metricPath + "." + k` since you don't need any of the conversion capabilities of Sprintf.